### ISSUE TITLE
Can’t compile: type mismatch resolving `<Vec<h3o::CellIndex> as IntoIterator>::Item == h3o::index::cell::CellIndex`

### ISSUE THREAD

**Issue by hfs** [HydroniumLabs/h3o-cli#3]
Can’t compile using

```bash
cargo install h3o-cli
```
with these error messages:

```
    Updating crates.io index
  Installing h3o-cli v0.1.1
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.56
   Compiling quote v1.0.26
   Compiling unicode-ident v1.0.8
   Compiling libc v0.2.142
   Compiling libm v0.2.6
   Compiling semver v1.0.17
   Compiling cfg-if v1.0.0
   Compiling scopeguard v1.1.0
   Compiling serde_derive v1.0.160
   Compiling crunchy v0.2.2
   Compiling byteorder v1.4.3
   Compiling num-traits v0.2.15
   Compiling lock_api v0.4.9
   Compiling rustc_version v0.4.0
   Compiling serde v1.0.160
   Compiling heapless v0.7.16
   Compiling syn v2.0.15
   Compiling proc-macro-hack v0.5.20+deprecated
   Compiling tiny-keccak v2.0.2
   Compiling spin v0.9.8
   Compiling hash32 v0.2.1
   Compiling io-lifetimes v1.0.10
   Compiling stable_deref_trait v1.2.0
   Compiling version_check v0.9.4
   Compiling once_cell v1.17.1
   Compiling getrandom v0.2.9
   Compiling rustix v0.37.17
   Compiling smallvec v1.10.0
   Compiling ahash v0.8.3
   Compiling approx v0.5.1
   Compiling bitflags v1.3.2
   Compiling rstar v0.9.3
   Compiling thiserror v1.0.40
   Compiling linux-raw-sys v0.3.6
   Compiling log v0.4.17
   Compiling utf8parse v0.2.1
   Compiling ieee754 v0.2.6
   Compiling radium v0.7.0
   Compiling const-random-macro v0.1.15
   Compiling konst_kernel v0.3.4
   Compiling memchr v2.5.0
   Compiling const_panic v0.2.7
   Compiling serde_json v1.0.96
   Compiling anstyle-parse v0.2.0
   Compiling konst v0.3.4
   Compiling accurate v0.3.1
   Compiling const-random v0.1.15
   Compiling either v1.8.1
   Compiling colorchoice v1.0.0
   Compiling anstyle v1.0.0
   Compiling itoa v1.0.6
   Compiling lazy_static v1.4.0
   Compiling ryu v1.0.13
   Compiling tap v1.0.1
   Compiling auto_ops v0.3.0
   Compiling anstyle-query v1.0.0
   Compiling float_eq v1.0.1
   Compiling wyz v0.5.1
   Compiling geographiclib-rs v0.2.3
   Compiling float_next_after v0.1.5
   Compiling robust v0.2.3
   Compiling clap_lex v0.4.1
   Compiling funty v2.0.0
   Compiling strsim v0.10.0
   Compiling anyhow v1.0.70
   Compiling is-terminal v0.4.7
   Compiling anstream v0.3.1
   Compiling heck v0.4.1
   Compiling clap_builder v4.2.5
   Compiling bitvec v1.0.1
   Compiling h3o v0.3.1
   Compiling quick-xml v0.22.0
   Compiling maplit v1.0.2
   Compiling thiserror-impl v1.0.40
   Compiling clap_derive v4.2.0
   Compiling serde_repr v0.1.12
   Compiling thc v0.1.2
   Compiling clap v4.2.5
   Compiling geo-types v0.7.9
   Compiling geo v0.23.1
   Compiling kml v0.6.0
   Compiling geojson v0.24.0
   Compiling h3o v0.2.0
   Compiling h3o-cli v0.1.1
error[E0271]: type mismatch resolving `<Vec<h3o::CellIndex> as IntoIterator>::Item == h3o::index::cell::CellIndex`
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/h3o-cli-0.1.1/src/commands/compress.rs:18:32
    |
18  |     thc::compress(&mut stdout, indexes).context("compression")?;
    |     -------------              ^^^^^^^ expected struct `h3o::index::cell::CellIndex`, found struct `h3o::CellIndex`
    |     |
    |     required by a bound introduced by this call
    |
    = note: perhaps two different versions of crate `h3o` are being used?
note: required by a bound in `compress`
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/thc-0.1.2/src/lib.rs:167:30
    |
167 |     cells: impl IntoIterator<Item = CellIndex>,
    |                              ^^^^^^^^^^^^^^^^ required by this bound in `compress`

error[E0277]: the trait bound `json::CellIndex: From<h3o::index::cell::CellIndex>` is not satisfied
  --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/h3o-cli-0.1.1/src/commands/decompress.rs:41:30
   |
41 |                 .map(|index| index.map(Into::into))
   |                              ^^^^^^^^^^^^^^^^^^^^^ the trait `From<h3o::index::cell::CellIndex>` is not implemented for `json::CellIndex`
   |
   = help: the trait `From<h3o::CellIndex>` is implemented for `json::CellIndex`
   = note: required for `h3o::index::cell::CellIndex` to implement `Into<json::CellIndex>`

error[E0277]: the trait bound `json::CellIndex: From<h3o::index::cell::CellIndex>` is not satisfied
  --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/h3o-cli-0.1.1/src/commands/decompress.rs:41:40
   |
41 |                 .map(|index| index.map(Into::into))
   |                                        ^^^^^^^^^^ the trait `From<h3o::index::cell::CellIndex>` is not implemented for `json::CellIndex`
   |
   = help: the trait `From<h3o::CellIndex>` is implemented for `json::CellIndex`
   = note: required for `h3o::index::cell::CellIndex` to implement `Into<json::CellIndex>`

Some errors have detailed explanations: E0271, E0277.
For more information about an error, try `rustc --explain E0271`.
error: could not compile `h3o-cli` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `h3o-cli v0.1.1`, intermediate artifacts can be found at `/tmp/cargo-installFv9q4g`
```

```
$ cargo --version                                                                                                                                                                                                                                               cargo 1.65.0
```

It looks like a dependency version conflict? `h3o` is included as `h3o v0.3.1` and `h3o v0.2.0`

Can you help me?

**Comment by hfs** [comment]
`cargo install --git https://github.com/HydroniumLabs/h3o-cli.git` works as expected, by the way, and is an easy workaround.

**Comment by grim7reaper** [comment]
Yeah, the last published version is a bit old...
I've just published a 0.1.2 to fix the issue.

Thanks for the heads up 🙂
